# HG changeset patch
# Parent 99164862ccce7947259465dcf9edbf6375d56413
# User Wolfgang Rosenauer <wr@rosenauer.org>
No bug - libproxy API change picked up in bmo#769764 but fails in openSUSE 11.2 and earlier. Still cast to char* to make "all" happy
Note: Remove that patch when openSUSE 11.2 runs out of any support!

diff --git a/toolkit/system/unixproxy/nsLibProxySettings.cpp b/toolkit/system/unixproxy/nsLibProxySettings.cpp
--- a/toolkit/system/unixproxy/nsLibProxySettings.cpp
+++ b/toolkit/system/unixproxy/nsLibProxySettings.cpp
@@ -67,17 +67,17 @@ nsUnixSystemProxySettings::GetProxyForUR
 
   if (!mProxyFactory) {
     mProxyFactory = px_proxy_factory_new();
   }
   NS_ENSURE_TRUE(mProxyFactory, NS_ERROR_NOT_AVAILABLE);
 
   char **proxyArray = nullptr;
   proxyArray = px_proxy_factory_get_proxies(mProxyFactory,
-                                            PromiseFlatCString(aSpec).get());
+                                            (char *) (PromiseFlatCString(aSpec).get()));
   NS_ENSURE_TRUE(proxyArray, NS_ERROR_NOT_AVAILABLE);
 
   // Translate libproxy's output to PAC string as expected
   // libproxy returns an array of proxies in the format:
   // <procotol>://[username:password@]proxy:port
   // or
   // direct://
   //
